onadata/apps/api/viewsets/user_profile_viewset.py

```diff
@@ -200,3 +204,64 @@ def monthly_submissions(self, request, *args, **kwargs):
 
         serializer = MonthlySubmissionsSerializer(instance_count, many=True)
         return Response(serializer.data[0])
+
+    @action(detail=False)
+    def verify_email(self, request, *args, **kwargs):
+        webhook = settings.EMAIL_VERIFICATION_WEBHOOK
+        redirect_url = settings.POST_EMAIL_VERIFICATION_REDIRECT_URL
+        verification_key = request.query_params.get('verification_key')
+        response_message = "Missing or invalid verification key"
+        if verification_key:
+            try:
+                rp = RegistrationProfile.objects.get(
+                    activation_key=verification_key
+                )
+            except RegistrationProfile.DoesNotExist:
+                pass
+            else:
+                response_message = "Email was NOT verified"
+                if webhook:
+                    payload = {
+                        'username': rp.user.username,
+                        'verified': True
+                    }
+                    r = requests.post(webhook, data=payload)
+                    if r.status_code == 200:
+                        rp.activation_key = settings.VERIFIED_KEY_TEXT
+                        rp.save()
+                        if redirect_url:
+                            return HttpResponseRedirect(redirect_url)
+
+                        response_message = "Email has been verified."
+
+                        return Response(response_message)
+
+        return HttpResponseBadRequest(response_message)
+
+    @action(methods=['POST'], detail=False)
+    def send_verification_email(self, request, *args, **kwargs):
```

Comment:
Not sure if I agree with the recommendation above. I can't move it outside the class because it's a class function, or a viewset action to be specific. If I'm not wrong, it's mainly complaining because i'm not using `self` object inside the function. I'd like to think the fix here would be retrieve the `request` object from `self` i.e. `self.request.user` but I don't see a plus to that since we already have the request object as part of the args. I'd like to assume that other actions in different classes use the request argument rather than retrieve it from self object, no?